configure.py

```diff
@@ -0,0 +1,408 @@
+        false_strings = ("0", "f", "false", "n", "no")
+        if var_content in true_strings:
+            var = True
+        elif var_content in false_strings:
+            var = False
+        else:
+            raise UserInputError(
+                "Environment variable %s must be set as a boolean indicator.\n"
+                "The following are accepted as TRUE : %s.\n"
+                "The following are accepted as FALSE: %s.\n"
+                "Current value is %s."
+                % (var_name, ", ".join(true_strings), ", ".join(false_strings), var)
+            )
+
+    while var is None:
+        user_input_origin = get_input(question)
+        user_input = user_input_origin.strip().lower()
+        if user_input == "y":
+            print(yes_reply)
+            var = True
+        elif user_input == "n":
+            print(no_reply)
+            var = False
+        elif not user_input:
+            if enabled_by_default:
+                print(yes_reply)
+                var = True
+            else:
+                print(no_reply)
+                var = False
+        else:
+            print("Invalid selection: {}".format(user_input_origin))
+    return var
+
+
+def run_shell(cmd, allow_non_zero=False, stderr=None):
+    if stderr is None:
+        stderr = sys.stdout
+    if allow_non_zero:
+        try:
+            output = subprocess.check_output(cmd, stderr=stderr)
+        except subprocess.CalledProcessError as e:
+            output = e.output
+    else:
+        output = subprocess.check_output(cmd, stderr=stderr)
+    return output.decode("UTF-8").strip()
+
+
+def cygpath(path):
+    """Convert path from posix to windows."""
+    return os.path.abspath(path).replace("\\", "/")
+
+
+def get_python_path(environ_cp, python_bin_path):
+    """Get the python site package paths."""
+    python_paths = []
+    if environ_cp.get("PYTHONPATH"):
+        python_paths = environ_cp.get("PYTHONPATH").split(":")
+    try:
+        stderr = open(os.devnull, "wb")
+        library_paths = run_shell(
+            [
+                python_bin_path,
+                "-c",
+                'import site; print("\\n".join(site.getsitepackages()))',
+            ],
+            stderr=stderr,
+        ).split("\n")
+    except subprocess.CalledProcessError:
+        library_paths = [
+            run_shell(
+                [
+                    python_bin_path,
+                    "-c",
+                    "from distutils.sysconfig import get_python_lib;"
+                    "print(get_python_lib())",
+                ]
+            )
+        ]
+
+    all_paths = set(python_paths + library_paths)
+
+    paths = []
+    for path in all_paths:
+        if os.path.isdir(path):
+            paths.append(path)
+    return paths
+
+
+def get_python_major_version(python_bin_path):
+    """Get the python major version."""
+    return run_shell([python_bin_path, "-c", "import sys; print(sys.version[0])"])
```

Comment:
The TF people probably have a reason for this, but since the script is already python, why could they not just `return sys.version[0]` here?

Reply:
I think it's because the version of Python that you're using to run this script is not necessarily the same version of Python that you're configuring for the build. You could have overriden the bin/lib paths to point to whatever random version of Python you like.

Reply:
Oh that makes sense, if you've set `PYTHON_BIN_PATH` from your shell. :+1: